### PR TITLE
R4R: Add flag --output-file to save result and ensure the result is consistent

### PR DESCRIPTION
**Pull request by chengwenxi**
close https://github.com/irisnet/irishub/issues/1060

refer to https://github.com/irisnet/irishub/issues/1040

### CODE REVIEW

File: server/export.go [hunk 1]
```diff
@@ -72,13 +74,25 @@ func ExportCmd(ctx *Context, cdc *codec.Codec, appExporter AppExporter) *cobra.C
...
-			encoded = sdk.MustSortJSON(encoded)
-			fmt.Println(string(encoded))
+
+			var outputRaw json.RawMessage
+			codec.Cdc.UnmarshalJSON(encoded, &outputRaw)
+			outputRaw = sdk.MustSortJSON(outputRaw)
+
+			output, err := codec.MarshalJSONIndent(cdc, outputRaw)
+
+			outputFile := viper.GetString(flagOutputFile)
```
Comment (wukongcheng) [inline server/export.go]: Should we mark this flagOutputFile flag as required?

Comment (chengwenxi) [reply server/export.go]: It have default value

**Comment by wukongcheng** [comment]
Also need to merge into develop branch

**Comment by chengwenxi** [comment]
Only appState has inconsistent objects, such as map, so only need to sort appState to make genesis.json more identical to the initial genesis.json.

File: server/export.go [hunk 2]
```diff
@@ -68,17 +69,25 @@ func ExportCmd(ctx *Context, cdc *codec.Codec, appExporter AppExporter) *cobra.C
 			doc.AppState = appState
 			doc.Validators = validators
 
+			doc.AppState = sdk.MustSortJSON(doc.AppState)
+
```
Comment (chengwenxi) [inline server/export.go]: Only appState has inconsistent objects, such as map, so only need to sort appState to make genesis.json more identical to the initial genesis.json.

**Pull request merged**